core/src/main/java/com/linecorp/armeria/common/HttpHeadersBase.java

```diff
@@ -113,13 +119,19 @@ class HttpHeadersBase
     HttpHeadersBase(HttpHeaderGetters parent) {
         super(parent);
         assert !(parent instanceof HttpHeadersBase);
+        contentLengthUnknown = parent.isContentLengthUnknown();
         endOfStream = parent.isEndOfStream();
         cache = new HashMap<>(4);
     }
 
     @Override
-    void onChange(@Nullable AsciiString name) {
+    void onChange(@Nullable AsciiString name, boolean add) {
+        if (HttpHeaderNames.CONTENT_LENGTH.equals(name) && add) {
+            contentLengthUnknown = false;
+        }
```

Comment:
I have a performance concern for this change because this method will be called for any header names. What do you think about changing like this:

- Remove this check in `onChange()`
- Rename `contentLengthUnknown` to `contentLengthMayBeUnknown`
- Modify `isContentLengthUnknown()` to check if `CONTENT_LENGTH` header exists:
  ```java
  public boolean isContentLengthUnknown() {
    if (contentLengthMayBeUnknown) {
      if (contains(CONTENT_LENGTH)) {
        return contentLengthMayBeUnknown = false;
      }
    } else {
      return false;
    }
  }
  ```

Reply:
Sounds like a good idea.